### PR TITLE
[13.0][IMP] sale_report_delivered: Discard customer returned stock moves that not to refund

### PR DESCRIPTION
**Pull request by sergio-teruel**
cc @Tecnativa TT34713

ping @carlosdauden @chienandalu

### CODE REVIEW

File: sale_report_delivered/reports/sale_report.py [hunk 1]
```diff
@@ -165,7 +165,7 @@ def _from(self):
 
     def _where(self):
         return """
-            WHERE sm.state = 'done' OR sm.state IS NULL
+            WHERE sm.to_refund = false AND (sm.state = 'done' OR sm.state IS NULL)
```
Comment (pedrobaeza) [inline sale_report_delivered/reports/sale_report.py]: For shorter form, but shouldn't be the other way (to include those to refund)?

```suggestion
            WHERE NOT sm.to_refund AND (sm.state = 'done' OR sm.state IS NULL)
```

**Comment by sergio-teruel** [comment]
@pedrobaeza review again please!!

File: sale_report_delivered/reports/sale_report.py [hunk 2]
```diff
@@ -165,7 +165,9 @@ def _from(self):
 
     def _where(self):
         return """
-            WHERE sm.state = 'done' OR sm.state IS NULL
+            WHERE (sm.state = 'done' OR sm.state IS NULL) AND
+                  (source_location.usage <> 'customer' AND
```
Comment (pedrobaeza) [inline sale_report_delivered/reports/sale_report.py]: Aren't you discarding with this all returns?

Comment (chienandalu) [reply sale_report_delivered/reports/sale_report.py]: Yes, maybe an easier fix could be to set the `signed_qty` to 0 for this special case

Comment (sergio-teruel) [reply sale_report_delivered/reports/sale_report.py]: I think it is better understood this way

**Review by carlosdauden — APPROVED** [review]

**Review by pedrobaeza — APPROVED** [review]

File: sale_report_delivered/reports/sale_report.py [hunk 3]
```diff
@@ -164,8 +164,15 @@ def _from(self):
...
     def _where(self):
+        """Take into account only stock moves from internal locations to other
+        locations and moves from customer with the field 'to_refund' True
+        """
         return """
-            WHERE sm.state = 'done' OR sm.state IS NULL
+            WHERE (sm.state = 'done' OR sm.state IS NULL) AND (
+                (source_location.usage = 'internal' AND dest_location.usage = 'customer') OR
+                (source_location.usage = 'customer' AND dest_location.usage = 'internal'
+                    AND sm.to_refund)
```
Comment (chienandalu) [inline sale_report_delivered/reports/sale_report.py:174]: This leaves out dropshipping, isn't it?

Comment (carlosdauden) [reply sale_report_delivered/reports/sale_report.py:174]: You're right.
A new PR could be made by changing both `usage = 'internal'` to `usage IN ('internal', 'supplier')`

**Pull request merged**